sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/AvroSchemaRegistryCodec.java

```diff
@@ -52,20 +45,18 @@ public SerializationType getSerializationType() {
      * @return avro schema
      */
     @Override
-    public Schema parseSchemaString(String schemaString) {
-        return (new Schema.Parser()).parse(schemaString);
+    public String parseSchemaString(String schemaString) {
```

Comment:
Not sure how this will work.

The output of this function is required for calling decode().  So if we don't do the parse here, then decode() will have to parse.  That'll lose the deserialization caching that happens on the SR client layer.

Reply:
OK I see.  So the provider will be in charge of caching.  That should be fine.

Reply:
SchemaRegistryObject has a getSchema() that will need to be removed.  It should only return a schema string now.